cachelib/navy/common/Device.cpp

```diff
@@ -1067,10 +1067,20 @@ IoContext* FileDevice::getIoContext() {
 }
 
 int FileDevice::allocatePlacementHandle() {
-  static constexpr uint16_t kDefaultFdpIdx = 0u;
 #ifndef CACHELIB_IOURING_DISABLE
-  if (fdpNvmeVec_.size() > 0) {
-    return fdpNvmeVec_[kDefaultFdpIdx]->allocateFdpHandle();
+  if (fdpNvmeDevs_.size() > 0) {
+    auto fdpHandle = -1;
+    //Ensuring that same FDP placement handle is allocated for all FdpNvme 
```

Comment:
> Asking to learn: I thought it would be fine if one RAID write gets written to different handles in different device. But because we don't want to keep track of a vector of different handles, we just make sure we have the same handle for all devices?

Yes, we wanted to avoid the code complexity and use a simple scheme of same handle for all the devices for a given placement handle. Having ability to write to different handles for different RAID devices could be nice but need to design carefully, and can be added later.